docs/user-guide/isis/sans2d.ipynb

```diff
@@ -47,97 +48,129 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4a2df47b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "workflow[OutFilename] = 'reduced.nxs'\n",
+    "\n",
+    "workflow[NeXusMonitorName[Incident]] = 'monitor2'\n",
+    "workflow[NeXusMonitorName[Transmission]] = 'monitor4'\n",
+    "\n",
+    "workflow[isis.SampleOffset] = sc.vector([0.0, 0.0, 0.053], unit='m')\n",
+    "workflow[isis.MonitorOffset[Transmission]] = sc.vector([0.0, 0.0, -6.719], unit='m')\n",
+    "\n",
+    "workflow[WavelengthBins] = sc.linspace(\n",
+    "    'wavelength', start=2.0, stop=16.0, num=141, unit='angstrom'\n",
     ")\n",
-    "providers = providers + (\n",
-    "    isis.data.transmission_from_background_run,\n",
-    "    isis.data.transmission_from_sample_run,\n",
-    "    sans.beam_center_from_center_of_mass,\n",
+    "\n",
+    "workflow[isis.sans2d.LowCountThreshold] = sc.scalar(100, unit='counts')\n",
+    "\n",
+    "mask_interval = sc.array(dims=['wavelength'], values=[2.21, 2.59], unit='angstrom')\n",
+    "workflow[WavelengthMask] = sc.DataArray(\n",
+    "    sc.array(dims=['wavelength'], values=[True]),\n",
+    "    coords={'wavelength': mask_interval},\n",
     ")\n",
     "\n",
-    "pipeline = sciline.Pipeline(providers=providers, params=params)"
+    "workflow[QBins] = sc.linspace(dim='Q', start=0.01, stop=0.6, num=141, unit='1/angstrom')\n",
+    "workflow[NonBackgroundWavelengthRange] = sc.array(\n",
+    "    dims=['wavelength'], values=[0.7, 17.1], unit='angstrom'\n",
+    ")\n",
+    "workflow[CorrectForGravity] = True\n",
+    "workflow[UncertaintyBroadcastMode] = UncertaintyBroadcastMode.upper_bound\n",
+    "workflow[ReturnEvents] = True"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "39f70669-0771-4a59-8e10-95c9120d0e9e",
+   "id": "67b18b0d",
    "metadata": {},
    "source": [
-    "## Visualize the pipeline\n",
+    "## Configuring data to load\n",
     "\n",
-    "Before we begin computations, we can visualize the pipeline:"
+    "We have not configured which files we want to load.\n",
+    "In this tutorial, we use a helper to get the tutorial data and set the corresponding `DataFolder` and filenames.\n",
+    "In a real use case, you would set these parameters manually:"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "d71780b8-56d5-445f-9d43-635d3d5f406b",
+   "id": "ff1658c5",
    "metadata": {},
    "outputs": [],
    "source": [
-    "# left-right layout works better for this graph\n",
-    "pipeline.visualize(BackgroundSubtractedIofQ, graph_attr={'rankdir': 'LR'})"
+    "workflow[DataFolder] = isis.data.get_sans2d_tutorial_data_folder()\n",
+    "workflow[DirectBeamFilename] = 'DIRECT_SANS2D_REAR_34327_4m_8mm_16Feb16.dat.h5'\n",
+    "workflow[Filename[SampleRun]] = 'SANS2D00063114.nxs.h5'\n",
+    "workflow[Filename[BackgroundRun]] = 'SANS2D00063159.nxs.h5'\n",
+    "workflow[Filename[EmptyBeamRun]] = 'SANS2D00063091.nxs.h5'"
```

Comment:
I'm not convinced by this approach. It is nice that `get_sans2d_tutorial_data_folder` places all files in the same folder and that we can then simply list the file names. But this pretends that the mechanism is flexible while in reality, the filenames are determined by the `get_*_data_folder` functions. So there is no need to list them separately.

If you want to use this as an example for users, this doesn't quite work either. Users would typically (I think) not provide a folder with all their files and list `Filename`s. Instead, they would, if they use local files directly, list `FilePath`s.

With SciCat, it would also not be possible to list file names after downloading because the names are required for the download itself.

Reply:
> If you want to use this as an example for users, this doesn't quite work either. Users would typically (I think) not provide a folder with all their files and list Filenames. Instead, they would, if they use local files directly, list FilePaths.

Yes, this was intended as an example for users. But this mechanism is what we have previously implemented, using the `DataFolder` mechanism. They can just set the to `'.'`, no?

Reply:
> With SciCat, it would also not be possible to list file names after downloading because the names are required for the download itself.

Isn't this a different purpose?
- For SciCat download, the user needs to list which files they need.
- For the workflow, they need to specify which file has which meaning (sample, background, ...).

So it is not clear to me if that duplication can be avoided? Even though one could imagine that download all files from an experiment, such that there is no need to list them individually for download?

Reply:
> Yes, this was intended as an example for users. But this mechanism is what we have previously implemented, using the DataFolder mechanism. They can just set the to '.', no?

Only if file paths are specified relative to CWD. And then it still goes through the path resolution `Filename` -> `FilePath`. I think it is cleanest to specify a `FilePath` directly to avoid any potential issues with having a bad resolver.


> Isn't this a different purpose?
> 
>     * For SciCat download, the user needs to list which files they need.
> 
>     * For the workflow, they need to specify which file has which meaning (sample, background, ...).
> 
> 
> So it is not clear to me if that duplication can be avoided? Even though one could imagine that download all files from an experiment, such that there is no need to list them individually for download?

The datasets should, at least partially, indicate the meaning of a file. E.g., a sample dataset should link to a background dataset, explicitly stating that this is a background. See https://github.com/scipp/essreduce/blob/5f23eaca66727f2507eb0d89896a947d106029d0/src/ess/reduce/scicat.py#L29 as proposed in https://github.com/scipp/essreduce/pull/18.

And conceptually, we also have a (`Filename`, `DatasetId`) -> `FilePath` resolver. So it is pretty much the same as in the pooch case except that it needs an additional input.

Reply:
### Conclusion from discussion:

We are considering to remove `DataFolder` and `FilePath`. These were previously added to "hide" differences between local files and files obtained via Pooch or Scicat. However, they are confusing to users, and complicate the visualized task graph.
Instead, use something like:

```python
# Note that this assumes the helper functions return sciline.Pipeline, which Pipeline.__setitem__
# can handle after the rewrite. Otherwise use `workflow.insert`.
workflow[Filename[SampleRun]] = download_tutorial_sample_data()
workflow[Filename[SampleRun]] = download_scicat_dataset(12345)
```

There are variations to this where a SciCat helper finds, e.g., background runs related to the specified sample run, such that those can be inserted into the workflow automatically.

There are some related questions around other meta data in the SciCat dataset that may be needed by the workflow. There are several solutions to this, but none of them appears to interfere with the decision to remove `DataFolder` and `FilePath`.

Reply:
@nvaytet Thoughts?

Reply:
I think it's great if we can remove `DataFolder` and `FilePath`.
I was wondering if the above could still be a bit confusing. If you read it twice, and are used to it, then it makes sense that `download_tutorial_sample_data()` would download the file somewhere, and then return the path to that file.
But the first time I read it, I sort of expected `download_tutorial_sample_data()` to return the actual data, not the `Filename`.
Can a slighty different (but not too lengthy) name help?